### PR TITLE
[Heartbeat] Add stringer redaction to synthexec cmd

### PR DESCRIPTION
**Pull request by emilioalvap**
## Proposed commit message

Redact synthetics cmd output to prevent logging of user-defined parameters.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Disruptive User Impact

N/A

## How to test this PR locally

1. Build heartbeat locally.
2. Add a browser monitor with any playwrightOpts or params.
3. Run and check logs show playwrightOpts or params as `REDACTED`

### CODE REVIEW

File: x-pack/heartbeat/monitors/browser/synthexec/synthcmd.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+package synthexec
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+)
+
+// Variant of exec.command with redacted params, which could contain sensitive stuff
```
Comment (vigneshshanmugam) [inline x-pack/heartbeat/monitors/browser/synthexec/synthcmd.go:14]:
```suggestion
// Variant of exec.command with redacted params and playwright options, which might contain sensitive information.  
```

File: x-pack/heartbeat/monitors/browser/synthexec/synthcmd.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+type SynthCmd struct {
+	*exec.Cmd
+}
+
+func (cmd *SynthCmd) String() string {
+	b := new(strings.Builder)
+	b.WriteString(cmd.Path)
+	for i := 1; i < len(cmd.Args); i++ {
```
Comment (vigneshshanmugam) [inline x-pack/heartbeat/monitors/browser/synthexec/synthcmd.go:22]: why not use range?

Comment (emilioalvap) [reply x-pack/heartbeat/monitors/browser/synthexec/synthcmd.go:22]: There's lookahead involved, we want to redact the parameter value following the key

Comment (vigneshshanmugam) [reply x-pack/heartbeat/monitors/browser/synthexec/synthcmd.go:23]: Ahh I see, missed that.

**Review by vigneshshanmugam — APPROVED** [review]
LGTM

**Pull request merged**